### PR TITLE
Populate indexName for BatchQuery

### PR DESCRIPTION
**Pull request by ykmr1224**
### Description
- Populate indexName for BatchQuery
- Create Index query with manual refresh now executed as Batch query, but batch query didn't populate indexName in AsyncQueryJobMetadata.

### Related Issues
n/a

### Check List
- [x] New functionality includes testing.
- [ ] New functionality has been documented.
 - [ ] New functionality has javadoc added.
 - [ ] New functionality has a user manual doc added.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md).
- [x] Commits are signed per the DCO using `--signoff`.
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/sql/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by vamsi-amazon** [comment]
How are we executing create manual indices earlier?

File: async-query-core/src/main/java/org/opensearch/sql/spark/dispatcher/BatchQueryHandler.java [hunk 1]
```diff
@@ -106,6 +106,13 @@ public DispatchQueryResponse submit(
         .resultIndex(dataSourceMetadata.getResultIndex())
         .datasourceName(dataSourceMetadata.getName())
         .jobType(JobType.INTERACTIVE)
+        .indexName(getIndexName(context))
```
Comment (vamsi-amazon) [inline async-query-core/src/main/java/org/opensearch/sql/spark/dispatcher/BatchQueryHandler.java:109]: As mentioned below, we can't have name in all kinds of flint index queries? What is the indexName used for?

Comment (ykmr1224) [reply async-query-core/src/main/java/org/opensearch/sql/spark/dispatcher/BatchQueryHandler.java:109]: DQS uses index name to populate their database to find the query where the index is created.

**Comment by ykmr1224** [comment]
> How are we executing create manual indices earlier?

It used to be handled by interactive job. It did not populate indexName as well.

**Review by seankao-az — DISMISSED** [review]
lgtm

**Review by vamsi-amazon — APPROVED** [review]

**Review by seankao-az — APPROVED** [review]

**Pull request merged**